app/general/tests/test_frontend.py

```diff
@@ -0,0 +1,60 @@
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+from selenium.common import TimeoutException
+from selenium.webdriver.chrome.webdriver import WebDriver, Options
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+class TestFrontend(StaticLiveServerTestCase):
+    @classmethod
+    def setUpClass(cls):
+        opts = Options()
+        opts.add_argument("--headless")
+        opts.add_argument("--no-sandbox")
+        opts.add_argument("--disable-dev-shm-usage")
+        opts.add_argument("--window-size=1920,1080")
+        cls.driver = WebDriver(opts)
+        cls.driver.implicitly_wait(5)
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.driver.quit()
+        super().tearDownClass()
+
+    def test_no_404s(self):
+        # Sanity check in case we ever change the 404 title
+        self.driver.get(f"{self.live_server_url}/blabla404")
+        print(self.driver.title)
+        assert self.driver.title.startswith(
+            "Error"
+        ), f"Actual title was {self.driver.title}. Page: {self.driver.page_source}"
+
+        # Check main page does not 404
+        self.driver.get(self.live_server_url)
+        self.assert_current_page_not_error()
+
+        # Check all nav items
+        for item in ["Search", "Institutions", "Projects", "Documents", "Languages", "Subjects"]:
+            self.check_nav_item(item)
+
+    def check_nav_item(self, link_text):
+        self.driver.find_element(By.PARTIAL_LINK_TEXT, link_text).click()
+
+        # We use 'in' to do a more permissive match (for instance 'Search' is usually '<search query> - Search'
+        try:
+            wait = WebDriverWait(self.driver, timeout=5)
```

Comment:
is it possible to use a global variable for timeout so it all tests can share the same max timeout?

Reply:
We can do this if desired - while we may not want all waits to be for the same amount of time (e.g. slow, bulk actions) we could always have multiple constants for different kinds of workloads